src/main/java/network/brightspots/rcv/ContestConfig.java

```diff
@@ -507,9 +507,16 @@ private void validateCvrFileSources() {
           isValid = false;
           Logger.log(
               Level.SEVERE,
-              "contestId is required for Dominion and Hart files.");
-        } else if (!isNullOrBlank(getContestId()) &&
-            !(provider == Provider.DOMINION || provider == Provider.HART)) {
+              String.format("contestId must be defined for CVR source with provider \"%s\"!",
+                  getProvider(source).toString()));
+        } else if (
+            !(provider == Provider.DOMINION || provider == Provider.HART) &&
+                fieldIsDefinedButShouldNotBeForProvider(
+                    getContestId(),
+                    "contestId",
+                    provider,
+                    source.getFilePath())
+        ) {
           isValid = false;
           Logger.log(
```

Comment:
Please delete this log message since it's already covered by `fieldIsDefinedButShouldNotBeForProvider`.

Reply:
oops, right